pkgs/applications/networking/cluster/waagent/default.nix

```diff
@@ -29,44 +22,33 @@ python39.pkgs.buildPythonPackage rec {
     sha256 = "0fvjanvsz1zyzhbjr2alq5fnld43mdd776r2qid5jy5glzv0xbhf";
   };
   patches = [
-    # Suppress the following error when waagent try to configure sshd:
+    # Suppress the following error when waagent tries to configure sshd:
     # Read-only file system: '/etc/ssh/sshd_config'
     ./dont-configure-sshd.patch
+
+    ./udev-set-mode-with-MODE.patch
   ];
   doCheck = false;
 
-  buildInputs = with python39.pkgs; [ distro ];
-  runtimeDeps = [
-    findutils
-    gnugrep
-    gnused
-    iproute2
-    iptables
-    nettools # for hostname
-    openssh
-    openssl
-    parted
-    procps # for pidof
-    shadow # for useradd, usermod
-    util-linux # for (u)mount, fdisk, sfdisk, mkswap
-  ];
+  propagatedBuildInputs = [ python.pkgs.distro ];
 
-  fixupPhase = ''
-     mkdir -p $out/bin/
-     WAAGENT=$(find $out -name waagent | grep sbin)
-     cp $WAAGENT $out/bin/waagent
-     wrapProgram "$out/bin/waagent" \
-         --prefix PYTHONPATH : $PYTHONPATH \
-         --prefix PATH : "${makeBinPath runtimeDeps}"
-     patchShebangs --build "$out/bin/"
+  # The binary entrypoint and udev rules are placed to the wrong place.
+  # Move them to their default location.
+  preFixup = ''
+    mv $out/${python.sitePackages}/usr/sbin $out/bin
+    rm $out/bin/waagent2.0
+    rmdir $out/${python.sitePackages}/usr
+
+    mv $out/${python.sitePackages}/etc $out/
```

Comment:
```suggestion
     ${python.pythonOnBuildForHost.interpreter} setup.py install_data --install-dir=$out --root=$out
```

Reply:
The proposed workaround in https://github.com/NixOS/nixpkgs/blob/fd7914c96f7c006047e0154dd239aa2396478094/doc/languages-frameworks/python.section.md?plain=1#L1684-L1689 doesn't work, because there's a lot more occurences. The entire file is 372 lines. I really think having 4 lines in fixup to move things into the right location is fine.